src/Adapter/Carrier/CommandHandler/AddCarrierHandler.php

```diff
@@ -63,10 +63,16 @@ public function handle(AddCarrierCommand $command): CarrierId
         $carrier->position = $command->getPosition();
         $carrier->active = $command->getActive();
         $carrier->delay = $command->getLocalizedDelay();
+        $carrier->max_width = $command->getMaxWidth();
+        $carrier->max_height = $command->getMaxHeight();
+        $carrier->max_weight = $command->getMaxWeight();
+        $carrier->max_depth = $command->getMaxDepth();
 
         $this->carrierValidator->validate($carrier);
 
         $carrierId = $this->carrierRepository->add($carrier);
+        $this->carrierValidator->validateGroupsExist($command->getGroupAccess());
```

Comment:
Similar to the comment in the edit handler, all the validation must be done at the beginning before anything is created or updated in the DB